src/norm/private/postgres/rowutils.nim

```diff
@@ -14,40 +15,47 @@ when (NimMajor, NimMinor) <= (1, 6):
 else:
   import std/macros
 
-proc fromRowPos[T: Model](obj: var T, row: Row, pos: var Natural, skip = false) =
+func isEmptyColumn*(row: Row, index: int): bool =
+  ## Checks whether the column at a given index is empty
+  row[index].kind == dvkNull
+
+## This does the actual heavy lifting for parsing
+proc fromRowPos[T: ref object](obj: var T, row: Row, pos: var Natural, skip: static bool = false) =
   ##[ Convert ``ndb.sqlite.Row`` instance into `Model`_ instance, from a given position.
 
   This is a helper proc to convert to `Model`_ instances that have fields of the same type.
   ]##
 
-  for fld, val in obj[].fieldPairs:
-    if val.isModel:                                 ## If we're dealing with a ``Model`` field
-      if val.model.isSome:                          ## and it's either a ``some Model`` or ``Model``
-        var subMod = get val.model                  ## then we try to populate it with the next ``row`` values.
+  for fld, dummyVal in T()[].fieldPairs:
```

Comment:
@PhilippMDoerner T() will not work for types that don't have default values for all.their fields.

Reply:
Can you maybe provide an example?
I've been wrecking my brain over a possible type that could break this and came up empty.
Not even having a field with the `DateTime` type would break this, since I never actually access the value inside the variable. I only ever check their types, which appears to get around such issues.

Reply:
I got one example courtesy of mratsim.
Any field of type `range(1..2)` (Or any other range that does not include null) will not work with this.
That... I'm not sure that is fixable. To iterate at compile time you need to be able to initialize at compile-time, that's pretty much where the 10% speed up stems from. And avoiding a whole bunch of procs just to please the compiler for no benefit.

Reply:
> Not even having a field with the DateTime type would break this

This is the example I was going to provide :-)

I'm afk now but I could swear T() would not compile if T is a ref object with a field of DateTime.

Also, since this proc accepts an instance anyway, why not use it? Why create a new one?

Reply:
That is because I went in with the goal to only compute at runtime what absolutely must be computed at runtime and move everything else to compile-time.
`T()` gives me an instance that exists at compile-time.  It is impossible to do the same with the object that is being passed because you only know that value at runtime. Strictly speaking `T()` *should* not be necessary, in terms of information it should suffice to just have an iterator over a type that spits out its field names and types, but nim doesn't have that so a compile-time instance is required. 

I make use of the for loop being compile-time in my various `when` statements in the code, these check at compile time whether the field you need to be parsing columns into is a primitive or a ref object or an optional of a primitive/ref object. Depending on that only the needed code for parsing this particular field is added to the loop.

This way you're achieving 2 things:
1) Computation gets outsourced into compile-time that you no longer have to do at runtime, speeding up the conversion slightly (which, as stated for my limited testing with models of 2-4 fields was roughly 10% faster than before)
2) No longer are procs needed only to please the compiler because invalid code-paths no longer exist at runtime. So the proc `func to*(dbVal; T: typedesc[Model]): T =` That you had to put in because the code-path that you land in line ` val = row[pos].to(typeof(val)) ` with val being a Model type exists, even though if-statements prevent that from ever being the case.

Reply:
> This is the example I was going to provide :-)

At least those I can guarantee you they work. Not only do I have plenty of fields in my side-project that work flawlessly with this version of norm (I'm currently using my own branch there since I wanted the `rawSelect` procs), the tests themselves would not work if DateTime objects caused a problem. E.g. `trelated.nim` contains testcases with models that have DateTime fields.